evmrpc/filter.go

```diff
@@ -1010,6 +1047,11 @@ func (f *LogFetcher) processBatch(ctx context.Context, start, end int64, crit fi
 		// Re-check cache after acquiring semaphore, in case another worker cached it.
 		if cachedEntry, found := f.globalBlockCache.Get(height); found {
 			<-f.dbReadSemaphore
+			if cachedEntry.Block != nil {
+				if err := f.watermarks.EnsureHeightAvailable(ctx, cachedEntry.Block.Block.Height); err != nil {
+					continue
+				}
+			}
```

Comment:
do we need to use `cachedEntry.Block.Block.Height` or should we consider just `height` that we have?

Reply:
This feels like logic that should have addressed the asked-for range well before we got this far? (like maybe pre-line 1028?)